---
title: "Summary Human"
description: |
  Overview of data analysis.
author:
  - name: Manuela Benary / Mohammad El-Ismail
    affiliation: CUBI / CCCC
    affiliation_url: 
date: "`r Sys.Date()`"
output: 
  distill::distill_article:
    toc: true
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---


```{r load-libraries}
library(Matrix)
library(Matrix.utils)
library(tidyverse)
library(Seurat)
library(dplyr)
library(knitr)
library(ggsci)
library(RCurl)
library(cowplot)
library(SingleR)
library(caret)
library(multtest)
library(clustree)
library(limma)
```

#Quality control metrices

```{r load-merged-seuratobject}
load("E://CCCC/PI3K/data/SeuratObjects/merged/Human/merged_seurat.RData")
```

```{r subset-seuratobject-by-sampletags}
merged_filtered_seurat <- subset(merged_seurat, subset = Sample_Tag != 'NA' & Sample_Tag != 'Undetermined' & Sample_Tag != 'SampleTag01_hs' & Sample_Tag != 'Multiplet')

merged_filtered_seurat$nCount_RNA <- NULL
merged_filtered_seurat$nFeature_RNA <- NULL
```

```{r}
merged_filtered_seurat_meta <- merged_filtered_seurat@meta.data
```

```{r subset-metadata-to-get-the-different-conditions}
subset_control <- subset(merged_filtered_seurat_meta, Sample_Tag == "Control" | Sample_Tag == "Control_tagged")
subset_alpelisib <- subset(merged_filtered_seurat_meta, Sample_Tag == "Alpelisib")
subset_copanlisib <- subset(merged_filtered_seurat_meta, Sample_Tag == "Copanlisib")
```

```{r Control-tags, fig.cap="The figures are showing different quality metrices for the control samples."}
#cell count
subset_control %>% 
  	ggplot(aes(x=Xenograft, fill=Xenograft)) + 
  	geom_bar() +
  	theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("NCells from the untreated xenografts")+
    scale_fill_lancet()

# Extract identity and sample information from dataframe to determine the number of cells per sample_tag
n_cells_control<- table(subset_control$Sample_Tag)

# View table
kable(n_cells_control,
      col.names = c("Sample_Tag",
                    "nCells"),
      caption = "The table shows the number of cells which were untreated.")

#UMI count
subset_control %>% 
  	ggplot(aes(color=Xenograft, x=nUMI, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
    scale_x_log10() + 
  	ylab("Cell density") +
  	geom_vline(xintercept = 500)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("nUMi from the untreated xenografts")+
    scale_fill_lancet()

#Gene Count
subset_control %>% 
  	ggplot(aes(color=Xenograft, x=nGene, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
  	scale_x_log10() + 
  	geom_vline(xintercept = 300)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("nGene from the untreated xenografts")+
    scale_fill_lancet()

#joint filtering effect
subset_control %>%
  	ggplot(aes(x=nUMI, y=nGene)) +
  	geom_point() +
	scale_colour_gradient(low = "gray90", high = "black") +
  	stat_smooth(method=lm) +
  	scale_x_log10() +
  	scale_y_log10() +
  	theme_classic() +
  	geom_vline(xintercept = 500) +
  	geom_hline(yintercept = 250) +
  	facet_wrap(~Sample_Tag)+
    scale_fill_lancet()

#Mitochondrial count
subset_control %>% 
  	ggplot(aes(color=Xenograft, x=percent.mt, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
  	scale_x_log10() +
  	geom_vline(xintercept = 2)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("Mitochondrial gene expression in [%] \n from the untreated xenografts")+
    scale_fill_lancet()
```

```{r Copanlisib-tag, fig.cap="The figures are showing different quality metrices for the copanlisib samples."}
#cell count
subset_copanlisib %>% 
  	ggplot(aes(x=Xenograft, fill=Xenograft)) + 
  	geom_bar() +
  	theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("NCells from the xenografts treated with Copanlisib")+
    scale_fill_lancet()

# Extract identity and sample information from dataframe to determine the number of cells per sample_tag
n_cells_copanlisib<- table(subset_copanlisib$Sample_Tag)

# View table
kable(n_cells_copanlisib,
      col.names = c("Sample_Tag",
                    "nCells"),
      caption = "The table shows the number of cells which were treated with Copanlisib.")

#UMI count
subset_copanlisib %>% 
  	ggplot(aes(color=Xenograft, x=nUMI, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
    scale_x_log10() +
  	ylab("Cell density") +
  	geom_vline(xintercept = 500)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("nUMI from the xenografts treated with Copanlisib")+
    scale_fill_lancet()

#Gene Count
subset_copanlisib %>% 
  	ggplot(aes(color=Xenograft, x=nGene, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) +
  	scale_x_log10() + 
  	geom_vline(xintercept = 300)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("nGene from the xenografts treated with Copanlisib")+
    scale_fill_lancet()

#joint filtering effect
subset_copanlisib %>%
  	ggplot(aes(x=nUMI, y=nGene)) +
  	geom_point() +
	scale_colour_gradient(low = "gray90", high = "black") +
  	stat_smooth(method=lm) +
  	scale_x_log10() +
  	scale_y_log10() +
  	geom_vline(xintercept = 500) +
  	geom_hline(yintercept = 250) +
  	facet_wrap(~Sample_Tag)+
    scale_fill_lancet()

#Mitochondrial count
subset_copanlisib %>% 
  	ggplot(aes(color=Xenograft, x=percent.mt, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
  	scale_x_log10() + 
  	geom_vline(xintercept = 2)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("Mitochondrial gene expression in [%] from \n the xenografts treated with Copanlisib")+
    scale_fill_lancet()
```

```{r Alpelisib-tag, fig.cap="The figures are showing different quality metrices for the alpelisib samples."}
#cell count
subset_alpelisib %>% 
  	ggplot(aes(x=Xenograft, fill=Xenograft)) + 
  	geom_bar() +
  	theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("NCells from the xenografts treated with Alpelisib")+
    scale_fill_lancet()

# Extract identity and sample information from dataframe to determine the number of cells per sample_tag
n_cells_alp<- table(subset_alpelisib$Sample_Tag)

# View table
kable(n_cells_alp,
      col.names = c("Sample_Tag",
                    "nCells"),
      caption = "The table shows the number of cells which were treated with Alpelisib.")

#UMI count
subset_alpelisib %>% 
  	ggplot(aes(color=Xenograft, x=nUMI, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
    scale_x_log10() +
  	ylab("Cell density") +
  	geom_vline(xintercept = 500)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("nUMI from the xenografts treated with Alpelisib")+
    scale_fill_lancet()

#Gene Count
subset_alpelisib %>% 
  	ggplot(aes(color=Xenograft, x=nGene, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) +
  	scale_x_log10() + 
  	geom_vline(xintercept = 300)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("nGene from the xenografts treated with Alpelisib")+
    scale_fill_lancet()

#joint filtering effect
subset_alpelisib %>%
  	ggplot(aes(x=nUMI, y=nGene)) +
  	geom_point() +
	  scale_colour_gradient(low = "gray90", high = "black") +
  	stat_smooth(method=lm) +
  	scale_x_log10() +
  	scale_y_log10() +
  	geom_vline(xintercept = 500) +
  	geom_hline(yintercept = 250) +
  	facet_wrap(~Sample_Tag)+
    scale_fill_lancet()

#Mitochondrial count
subset_alpelisib %>% 
  	ggplot(aes(color=Xenograft, x=percent.mt, fill=Xenograft)) + 
  	geom_density(alpha = 0.2) + 
  	scale_x_log10() + 
  	geom_vline(xintercept = 2)+
    theme(plot.title = element_text(hjust=0.5, face="bold")) +
  	ggtitle("Mitochondrial gene expression in [%] from \n the xenografts treated with Alpelisib")+
    scale_fill_lancet()
```

```{r}
merged_filtered_seurat_meta %>% 
    ggplot(aes(x=Xenograft, fill=nUMI)) + 
    geom_bar() +
    labs(x="Xenografts", y="Number of cells",
               title="Overview of the number of cells by xenografts and condition") +
    facet_grid(~Sample_Tag) +
    theme(axis.text.x = element_text(angle=45, vjust=1, hjust=1)) +
    scale_fill_lancet()
```



#Investigation of unwanted variations - cell cycle phases and mitochondrial gene expression

```{r}
load("E://CCCC/PI3K/data/SeuratObjects/filtered/Human/filtered_seurat.RData")
filtered_seurat$nCount_RNA <- NULL
filtered_seurat$nFeature_RNA <- NULL
filtered_seurat$orig.ident <- NULL
```

```{r load-cell-cycle-markers}
load("E://CCCC/PI3K/data/SuplementaryFiles/SeuratObjects/cycle.rda")
```

```{r Score cells for cell cycle}
filtered_phase_seurat <- CellCycleScoring(filtered_seurat, 
                                 g2m.features = g2m_genes, 
                                 s.features = s_genes)
```

```{r Identify the most variable genes}
filtered_phase_seurat <- FindVariableFeatures(filtered_phase_seurat,
                                     selection.method = "vst",
                                     nfeatures = 2000,
                                     verbose = FALSE)
```

```{r scale the counts}
filtered_phase_seurat <- ScaleData(filtered_phase_seurat)
```

```{r Perform PCA}
filtered_phase_seurat <- RunPCA(filtered_phase_seurat)
```

```{r Plot the PCA colored by cell cycle phase}
DimPlot(filtered_phase_seurat,
        reduction = "pca",
        group.by= "Phase",
        split.by = "Phase")+
  ggtitle("PCA visualization of the cell cycle phases")+
  scale_fill_lancet()
```

```{r}
filtered_phase_seurat <- RunUMAP(filtered_phase_seurat,
                             dims = 1:40,
                             reduction = "pca")

```

```{r}
DimPlot(filtered_phase_seurat,
        reduction = "umap",
        split.by = "Phase",
        group.by = "Sample_Tag") + 
  ggtitle("UMAP splitted by cell cycle phase \n and grouped by conditions") + 
  scale_fill_lancet()
```
```{r}
FeaturePlot(filtered_phase_seurat,
            features = "percent.mt")+
   ggtitle("Mitochondrial gene expression distribution in clusters")+
   scale_fill_lancet()
```
#Integration

```{r}
integrated_filtered_sct_cond_seurat <- readRDS("E://CCCC/PI3K/data/SeuratObjects/integrated/Human/integrated_filtered_sct_cond_seurat.rds")
```

```{r Visualization}
p1 <- DimPlot(integrated_filtered_sct_cond_seurat, reduction = "umap", group.by = "Sample_Tag") +
      labs(title = "UMAP visualisation of the condition distribution") +
      scale_fill_lancet()
p2 <- DimPlot(integrated_filtered_sct_cond_seurat, reduction = "umap", label = TRUE, repel = TRUE)+
      labs(title = "UMAP visualisation of the number of clusters") +
      scale_fill_lancet()
p3 <- DimPlot(integrated_filtered_sct_cond_seurat, reduction = "umap", group.by = "Xenograft")+
      labs(title = "UMAP visualisation of the xenograft distribution") +
      scale_fill_lancet()
p1 
p2 
p3 
```
#Clustering analysis

```{r}
seurat_integrated <- readRDS("E://CCCC/PI3K/data/SeuratObjects/clusteranalysis/Human/seurat_integrated_newidents.rds")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.1"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6) +
        labs(title = "UMAP visualisation shows the number of clusters \n for resolution 1.0") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.0.2"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 0.2") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.0.4"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 0.4") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.0.6"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 0.6") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.0.8"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 0.8") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.1.4"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 1.4") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.1.6"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 1.6") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.1.8"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 1.8") +
        scale_fill_lancet(name="ClusterIDs")
```

```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.2"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 2.0") +
        scale_fill_lancet(name="ClusterIDs")
```


```{r}
# Assign identity of clusters
Idents(object = seurat_integrated) <- "integrated_snn_res.2.4"
```

```{r}
# Plot the UMAP
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)+
        labs(title = "UMAP visualisation shows the number of clusters \n for the resolution 2.4") +
        scale_fill_lancet(name="ClusterIDs")
```



#Celltype identification singleR

```{r}
seurat_integrated_renameidents <- readRDS("E://CCCC/PI3K/data/SeuratObjects/celltypeIdentification/Human/SingleR/seurat_integrated_singleR.rds")
```

```{r}
DimPlot(object = seurat_integrated_renameidents, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE) +
        labs(title = "UMAP visualisation shows the different celltypes \n for the resolution 0.4") +
        scale_fill_lancet(name="Celltypes")
```

```{r}
n_cells0.4.long <- readRDS("E://CCCC/PI3K/data/SeuratObjects/celltypeIdentification/Human/SingleR/n_cells0.4.rds")
```

```{r}
plt_0.4 <- ggplot(n_cells0.4.long,aes(ClusterID, singlr_labels)) +
  geom_tile(aes(fill=Occurances)) +  
  geom_text(aes(label = round(Occurances, 1)), size=2) +
  scale_fill_gradient(low = "white", high = "red") +
  labs(y="Celltypes")

plt_0.4
```


#Label comparison

```{r}
df_merged <- readRDS("E://CCCC/PI3K/data/SeuratObjects/celltypeIdentification/df_merged_singleR_labels.rds")

plt_merged <- ggplot(df_merged,aes(x = singlr_labels.human, y= singlr_labels.mouse)) +
  geom_tile(aes(fill=Cell_Index_count)) +
  geom_text(aes(label = Cell_Index_count), size=1.5) +
  scale_fill_gradient(low = "white", high = "red") +
  labs(x="Celltypes_human", y="Celltypes_mouse") +
  theme(axis.text.x = element_text(angle=90))

plt_merged
```

#Celltype identification puram

```{r}
seurat_integrated_renamedIdents <- readRDS("E://CCCC/PI3K/data/SeuratObjects/celltypeIdentification/Human/Puram/seurat_integrated_puram.rds")
```

```{r}
DimPlot(object = seurat_integrated_renamedIdents, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE) +
        labs(title = "UMAP visualisation shows the different celltypes \n for the resolution 0.4") +
        scale_fill_lancet(name="Celltypes")
```

```{r}
n_cells0.4.long <- readRDS("E://CCCC/PI3K/data/SeuratObjects/celltypeIdentification/Human/Puram/n_cells0.4.rds")
```

```{r}
plt_0.4 <- ggplot(n_cells0.4.long,aes(ClusterID, predicted.id)) +
  geom_tile(aes(fill=Occurances)) +  
  geom_text(aes(label = round(Occurances, 1)), size=2) +
  scale_fill_gradient(low = "white", high = "red") +
  labs(y="Celltypes")

plt_0.4
```


#Transcript analysis

```{r}
umi_counts_merged <- read_delim("E://CCCC/PI3K/data/SuplementaryFiles/CSVFiles/umi_count_merged.csv")
```

```{r}
#create scatter plot
sp <- ggplot(umi_counts_merged, aes(x = UMIs_human, y = UMIs_mouse, fill=Labels)) + geom_point(aes(colour = Labels))
sp + geom_abline() +
  xlab("Transcripts human") +
  ylab("Transcripts mouse")
```



#Celltype identification demultiplexed

```{r load-data}
seurat_integrated <- readRDS("E://CCCC/PI3K/data/SeuratObjects/cellfiltering_umi/Human/seurat_integrated_with_spec_labels.rds")
seurat_integrated_filtered_human <- readRDS("E://CCCC/PI3K/data/SeuratObjects/demultiplexed/Human/seurat_integrated_filtered_human.rds")
```

```{r}
DimPlot(seurat_integrated, reduction = "umap", group.by= "Labels", cols = c("red","lightgrey","purple")) + ggtitle("UMAP human cells")
DimPlot(seurat_integrated, reduction = "umap", group.by = "Sample_Tag") + ggtitle("UMAP human cells by conditions")
```

```{r}
seurat_integrated_filtered_human <- RunUMAP(seurat_integrated_filtered_human, dims = 1:30)
DimPlot(seurat_integrated, reduction = "umap", group.by= "Labels", cols = c("red","lightgrey","purple")) + ggtitle("UMAP filtered human cells")
DimPlot(seurat_integrated_filtered_human, reduction = "umap", group.by = "Sample_Tag") + ggtitle("UMAP filtered human cells by conditions")
```

```{r select-xenograft-and-labels-from-metadata}
#selects the xenograft and labels columns from the metadata 
metadata <- seurat_integrated@meta.data
metadata %>% select(2,26) -> metadata
table(metadata)
```

```{r}
#generates the counts and probabilities for each Label/Xenograft
prop.table(table(metadata))
```

```{r load-data}
seurat_integrated_renamedIdents <- readRDS("E://CCCC/PI3K/data/SeuratObjects/cellfiltering_umi/Human/Puram/seurat_integrated_filtered_puram.rds")
```

```{r}
DimPlot(object = seurat_integrated_renamedIdents, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE) +
        labs(title = "UMAP visualisation shows the different celltypes \n for the resolution 0.4") +
        scale_fill_lancet(name="Celltypes")
```

```{r}
n_cells0.4.long <- readRDS("E://CCCC/PI3K/data/SeuratObjects/celltypeIdentification/Human/Puram/n_cells0.4.rds")
```

```{r}
plt_0.4 <- ggplot(n_cells0.4.long,aes(ClusterID, predicted.id)) +
  geom_tile(aes(fill=Occurances)) +  
  geom_text(aes(label = round(Occurances, 1)), size=2) +
  scale_fill_gradient(low = "white", high = "red") +
  labs(y="Celltypes")

plt_0.4
```