---
title: "Single cell sequencing of PDX models"
description: |
  Overview of data analysis for the pilot samples.
author:
  - name: Manuela Benary / Mohammad El-Ismail
    affiliation: CUBI / CCCC
    affiliation_url: 
date: "`r Sys.Date()`"
bibliography: literature.bib
output: 
  distill::distill_article:
    toc: true
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, cache = TRUE)
options(scipen = 999)
```

```{r load-libraries}
library(Matrix)
library(Matrix.utils)
library(tidyverse)
library(Seurat)
library(janitor)
library(ggsci)
library(ggpubr)
library(cowplot)
library(DT)
library(biomaRt)
source("helperFunctions.R")
```

- [ ] Translate German text into English
- [ ] Create Seurat Objects all samples with human annotation vs mouse annotation (4 Seurat objects)
- [ ] Show overview table of annotation-structure (gene types and numbers) for BD Rhapsody vs CHR (here also mouse vs human) -> section "Annotation and Mapping"
- [ ] Create figure of position-based GC content for both samples -> we would like to see whether the AT-content is enriched at the ends of the reads
- [ ] Create comparative table of the 4 annotations ->  wide format with columns for each sample and each annotation (Section Quality Filters)
- [ ] Prepare overview figure for molecular index statistics -> facet_grid (annotation ~ sample)
- [ ] Prepare overview figure for Cell label filtering with similar structure of molecular index statistics
- [ ] Add legends to all figures and tables -> use chunk-options
- [ ] Read the paper from Aznaourova et al (https://doi.org/10.1101/2021.11.05.467458) and summarize relevant method points for the analysis of lncRNA using BD Rhapsody

```{r load-data, eval=FALSE}
load("data/unfilteredSeurat_full.Rdata")
lapply(list_seurat, function(x) dim(x) %>% enframe(name = "dim")) %>% 
  bind_rows(.id = "experiment") %>% 
  mutate(dim = ifelse(dim == 1, "nrGenes", "nrCells")) %>% 
  mutate(filter = "none") -> countCells

mypal <- pal_jama()(2)

```

# Data overview {#top}

1. HNSCC-PDX (untreated, 1820): 
    - The probe was separated after dissociation. One part was marked with a sample tag, the other was unlabeled. Both probes were loaded on the cartridge and translated into cDNA. The libraries were prepared for the sample tag as well as the target cDNA. They both have the same index.
    - This approach was used to test the specifity of the sample tags.

1. HNSCC-PDX (treated with alpelisib, 1822): 
    - Right after dissociation the probe was loaded on the cartridge. This probe has no sample tag and has a different index. 

# Material and Methods

All probes have a spike-in with PhiX before the sequencing at the BIH Core Facility Genomics.

## Sequencing

Cells were sequenced with the BD Rhapsody System. This is a cartridge-based system, where single cells are loaded and lysed. During lysis RNA is coupled to beads and is subsequently reverse-transcribed into cDNA. The system allows to multiplex different samples, where each sample is marked with a specific sample tag. With this system it is also possible to analyse protein expression on a single cell level using antibody based sequencing.

## Alignment and Demultiplexing

- bbsplit  (using mouse and human) 
  - so far no successful run possible
- demultiplexing and alignment to human reference genome GrCH38 was done using 7bridges

-> back to [top][Data overview]

# Question and Tasks

The major question in this project is how does Alpelisib-treatment change signaling in HNSCC. 

Before we can do a signaling analysis, we have to solve a couple of technical tasks, which are detailed below:

1. Warum gehen so viele Reads verloren? 
    - das Trimming auf R1 ist nicht notwendig und evtl können so einige Reads erhalten bleiben
    - die Parameter von Star-Aligner können noch optimiert werden
2. Der GC-Gehalt ist geringer (42%) als erwartet - warum?
    - Spike-In mit PhiX -> allerdings hat PhiX ein GC-Gehalt von 47%, das ist wahrscheinlich nicht der Grund
    - die Kontamination mit PhiX ist ca 2% und sollte beim Alignment kein Problem darstellen
    - aufgrund des nekrotischen Materials könnte mtRNA enriched sein -> aber auch hier liegt der GC-Gehalt theoretisch bei ~48% und ist wahrscheinlich nicht ausschlaggebend
    - es könnte ein zelltyp-spezifischer Effekt sein -> sind bei HNC-Tumoren bestimmte RNAs überexprimiert?
    - Nebenbaustelle: Der GC-Gehalt ist positionsabhängig mit einer Änderung an Position 8-9 -> woran liegt das?
    - Struktur der Libraries ist PolyA -> wird das wirklich rausgeschnitten -> falls nicht, sollte es ein Enrichment an den Enden geben
3. Wieviel % der Samples sind maus-spezifisch?

    - Meines Wissens können PDX-Modelle bis zu 70-80\% stromal cells von Maus enthalten - Gibt es für Eure Modelle dafür Abschätzungen? Bei der RNA-Sequenzierung der Xenograft-abgeleiteten Organoide von der Peritonealkarzinose (Mathias ->EFRE-Projekt) wurde bis zu 80% Mausanteil in der Probe festgestellt. Mathias hat basierend der Daten vermutet, dass  der Anteil der Stromazellen von der Maus pro Passage zunimmt.  
    - Vergleich von tagged sequences mit verschiedenen Alignment-Ansätzen.
    - die aktuellen Parameter für das Alignment sind soweit unproblematisch
    - unabhängiges Alignment auf Maus bzw Mensch suggeriert:
        - Sample 1820 hat 18% human  vs 61% mouse
        - Sample 1822 hat 20% human  vs 64% mouse
4. Wieviele Zellen müssen am Anfang für die Sequenzierung genutzt werden?
    - abhängig von:
        - wieviele doublets / ungemappte Zellen fallen raus?
        - wieviele sterbende / tote Zellen fallen raus?
        - wieviel Zellen sind maus-spezifisch?
        - gibt es in der unbehandelten Probe Cluster bzw Subtypen von Zellen, die berücksichtigt werden müssen?
        
-> back to [top][Data overview]

# Analysis of pipeline results from 7bridges
## Experimental design

Changing the number of cells rather than the sequencing depth might be an option. See https://www.nature.com/articles/s41467-020-14482-y

Changing cell number needs to be done before library preparation.

## Annotation and Mapping
## Quality Filters
```{r bdQuality, layout="l-page"}
fname <- "data/1820_Metrics_Summary.csv"
readMetricsSummary(fname) %>% 
  mutate(genome = "human") %>% 
  bind_rows(readMetricsSummary("data/mouse_run/_1_1820_Metrics_Summary.csv") %>% mutate(genome = "mouse")) %>% 
  filter(!is.na(value)) -> metrics

metrics %>% 
  datatable()
```

## Molecular index statistics

PCR and sequencing errors within UMI sequence generate artifact molecules and contribute to over estimation of molecules. Molecular Index (MI) adjustment identify these artifacts and discard or re-assign them to their original index (from BD slides).

```{r umiStatistics}
read_csv("data/1820_UMI_Adjusted_Stats.csv", comment = "##") %>% 
  mutate(genome = "human") %>% 
  bind_rows(
    read_csv("data/mouse_run/_1_1820_UMI_Adjusted_Stats.csv", comment = "##") %>% 
      mutate(genome = "mouse")) -> umi.stats

umi.stats %>% 
  group_by(genome) %>% 
  summarize(meanRSEC = mean(RSEC_Adjusted_Seq_Depth_without_Singletons))

umi.stats %>% 
  gghistogram(x = "RSEC_Adjusted_Seq_Depth_without_Singletons", fill = "Status", facet.by = "genome") + 
  scale_x_log10() + 
  geom_vline(xintercept = 6, color = "darkred", lty = "dashed", size = 1)
```

In general, this is a two-step process, including RSEC-adjustment and DBEC-adjustment

1. Recursive Substitution Error Correction (RSEC)

- Corrects for MI errors that are derived from sequencing base calls and PCR substitution errors
- Based on edit distance of molecular indices
- Similar method used by most other competitors/academic pipelines

2. Distribution Based Error Correction (DBEC)

- Adjusts for errors derived from library preparation steps or sequencing base deletions.
- Based on distribution of sequencing coverage of molecular indices
- Developed at BD Genomics

Due to the low sequencing depth in our example (see threshold in Figure), DBEC-adjustment was not applied.

## Cell label filtering

The number of unique cell labels should be similar to the number of cells captured and amplified by the BD Rhapsody workflow. Causes of excess cell labels (from BD slides) :

- Spill over of mRNA to the neighboring wells
- Underloading beads in Rhapsody Cartridges: cells without beads and the RNA from the cells diffusing to adjacent wells
- Experiencing low level contamination during oligonucleotide and bead synthesis
- Generating errors during the PCR amplification steps of the workflow

In our case, the analysis of cell labels indicates two cell populations with very different expression levels:

![](data/cell_labeling_inflection.png)

