---
title: "Single cell sequencing of PDX models"
description: |
  Overview of data analysis for the pilot samples.
author:
  - name: Manuela Benary / Mohammad El-Ismail
    affiliation: CUBI / CCCC
    affiliation_url: 
date: "`r Sys.Date()`"
bibliography: literature.bib
output: 
  distill::distill_article:
    toc: true
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, cache = TRUE)
options(scipen = 999)
```

```{r load-libraries}
library(Matrix)
library(Matrix.utils)
library(tidyverse)
library(rtracklayer)
library(Seurat)
library(knitr)
library(janitor)
library(ggsci)
library(ggpubr)
library(cowplot)
library(DT)
library(biomaRt)
source("helperFunctions.R")
```

- [x] Translate German text into English
- [x] Create Seurat Objects all samples with human annotation vs mouse annotation (4 Seurat objects)
- [x] Show overview table of annotation-structure (gene types and numbers) for BD Rhapsody vs CHR (here also mouse vs human) -> section "Annotation and Mapping"
- [x] Create figure of position-based GC content for both samples -> we would like to see whether the AT-content is enriched at the ends of the reads
- [x] Create comparative table of the 4 annotations ->  wide format with columns for each sample and each annotation (Section Quality Filters)
- [x] Prepare overview figure for molecular index statistics -> facet_grid (annotation ~ sample)
- [x] Prepare overview figure for Cell label filtering with similar structure of molecular index statistics
- [x] Add legends to all figures and tables -> use chunk-options
- [ ] Read the paper from Aznaourova et al (https://doi.org/10.1101/2021.11.05.467458) and summarize relevant method points for the analysis of lncRNA using BD Rhapsody
- [ ] Adjust section on Normalization for the comparison between mouse and human -> update when finished
- [ ] Upload data to irods/SODAR (fastq/bam/multiqc)
- [ ] Run pipeline for 7270
- [ ] Run pipeline for 7271
- [ ] Run pipeline for 7272
- [ ] Run pipeline for 7273

```{r load-data, eval=FALSE}
load("data/unfilteredSeurat_1820_human.Rdata")
load("data/unfilteredSeurat_1822_human.Rdata")
load("data/unfilteredSeurat_1820_mouse.Rdata")
load("data/unfilteredSeurat_1822_mouse.Rdata")

lapply(list_seurat, function(x) dim(x) %>% enframe(name = "dim")) %>% 
  bind_rows(.id = "experiment") %>% 
  mutate(dim = ifelse(dim == 1, "nrGenes", "nrCells")) %>% 
  mutate(filter = "none") -> countCells

mypal <- pal_jama()(2)

```

# Data overview {#top}

1. HNSCC-PDX (untreated, 1820): 
    - The probe was separated after dissociation. One part was marked with a sample tag, the other was unlabeled. Both probes were loaded on the cartridge and translated into cDNA. The libraries were prepared for the sample tag as well as the target cDNA. They both have the same index.
    - This approach was used to test the specifity of the sample tags.

1. HNSCC-PDX (treated with alpelisib, 1822): 
    - Right after dissociation the probe was loaded on the cartridge. This probe has no sample tag and has a different index. 

# Material and Methods

All probes have a spike-in with PhiX before the sequencing at the BIH Core Facility Genomics.

## Sequencing

Cells were sequenced with the BD Rhapsody System. This is a cartridge-based system, where single cells are loaded and lysed. During lysis RNA is coupled to beads and is subsequently reverse-transcribed into cDNA. The system allows to multiplex different samples, where each sample is marked with a specific sample tag. With this system it is also possible to analyse protein expression on a single cell level using antibody based sequencing.

## Alignment and Demultiplexing

- bbsplit  (using mouse and human) 
  - so far no successful run possible
- demultiplexing and alignment to human reference genome GrCH38 was done using 7bridges

-> back to [top][Data overview]

# Question and Tasks

The major question in this project is how does Alpelisib-treatment change signaling in HNSCC. 

Before we can do a signaling analysis, we have to solve a couple of technical tasks, which are detailed below:

1. Why do we lose a lot of Reads ? 
    - trimming of R1 is not necessary and possibly some reads can be preserved this way
    - the parameters of the STAR- Aligner can still be optimized
2. GC content is lower than expected (42%) - why?
    - spike-in with PhiX -> however PhiX has a GC content of 47%, this is probably not the reason
    - contamination with PhiX is about 2% and should not be a problem for alignment
    - due to the necrotic material mtRNA could be enriched -> but also here the GC content is theoretically ~48% and is probably not crucial
    - it could be a cell type specific effect -> are certain RNAs overexpressed in HNC tumors?
    - Minor point: GC content is position dependent with a change at position 8-9 -> what is the reason for this?
    - Structure of libraries is polyA -> is this really edited out -> if not, there should be enrichment at the end
    
```{r fig.cap= "GC Content plot from the 1820 sample."}
include_graphics("plots/fastqc_per_sequence_gc_content_plot_1820.png")
```

```{r fig.cap= "GC Content plot from the 1822 sample.", layout="1-page"}
include_graphics("plots/fastqc_per_sequence_gc_content_plot_1822.png")
```

3. How many % of the samples are mouse specific?
    - to my knowledge PDX models can contain up to 70-80\% mouse stromal cells - Are there any estimates for this for your models? RNA sequencing of xenograft-derived organoids from peritoneal carcinomatosis (Mathias ->EFRE project) found up to 80% mouse content in the sample. Mathias hypothesized based on the data that the proportion of stromal cells from mouse increases per passage.  
    - comparison of tagged sequences with different alignment approaches.
    - the current parameters for the alignment are unproblematic so far
    - independent alignment on mouse or human suggests, using the annotation file provided by 7bridges:
        - sample 1820 has 18% human vs 61% mouse
        - sample 1822 has 20% human vs 64% mouse
    - independent alignment on human suggests, using the long non-coding annotation file:
        - sample 1820 has 37% human 
        - sample 1822 has 40% human 
    - independent alignment on mouse or human suggests, using the comprehensive annotation file:
        - sample 1820 has 41% human vs 70% mouse
        - sample 1822 has 42% human vs 71% mouse
4. How many cells must be used for sequencing at the beginning?
    - depending on:
        - how many doublets / unmapped cells fall out?
        - how many dying / dead cells fall out?
        - how many cells are mouse-specific?
        - are there clusters or subtypes of cells in the untreated sample that need to be considered?
        
-> back to [top][Data overview]

# Analysis of pipeline results from 7bridges
## Experimental design

Changing the number of cells rather than the sequencing depth might be an option. See https://www.nature.com/articles/s41467-020-14482-y
Changing cell number needs to be done before library preparation.

## Annotation and Mapping
To create the index files for the mapping step with the STAR-Aligner we used three different annotation files per sample and genome (Human & Mouse).
The first annotation file was provided by 7bridges.
PRI - It contains the comprehensive gene annotation on the primary assembly (chromosomes and scaffolds) sequence regions
    - This is a superset of the main annotation file
The next annotation file was used because we are thinking that long non-coding RNA is playing an important role in HNSCC.
NLC - It contains the comprehensive gene annotation of lncRNA genes on the reference chromosomes
    - This is a subset of the main annotation file
The third annotation file was used because we did-not found any mtRNA in the long non-coding RNA annotation file. Thats why we used the comprehensive gene annotation file which contains gene types from both other annotations.
CHR - It contains the comprehensive gene annotation on the reference chromosomes only
    - This is the main annotation file for most users
The tables below show the three annotations files for human and mouse. It contains the different gene types and their number in each of the files. 

```{r layout="l-page"}
## load data
gtf_nlc <- as.data.frame(import("data/Gtf_Annotation/Human/gencode.v29.long_noncoding_RNAs.gtf"))%>% 
  dplyr::select(gene_type) %>% 
  count(gene_type)

gtf_pri <- as.data.frame(import("data/Gtf_Annotation/Human/gencode.v29.primary_assembly.annotation.gtf")) %>%
  dplyr::select(gene_type) %>% 
  count(gene_type)
  
gtf_chr <- as.data.frame(import("data/Gtf_Annotation/Human/gencode.v29.annotation.gtf")) %>%
  dplyr::select(gene_type) %>% 
  count(gene_type)


## combine dataframes to one table
table_gtfs <- full_join(gtf_nlc, gtf_pri, by = 'gene_type', all=TRUE) %>%
  full_join(., gtf_chr, by = 'gene_type', all=TRUE)

names(table_gtfs) <- c("Gene_Type", "gtf_nlc", "gtf_pri", "gtf_chr")

knitr::kable(table_gtfs,
             caption = "The table shows the three different human annotation files with the gene types they contain")

```


```{r layout="l-page"}
## load data
nlc_mouse <- as.data.frame(import("data/Gtf_Annotation/Mouse/gencode.vM19.long_noncoding_RNAs.gtf"))%>% 
  dplyr::select(gene_type) %>% 
  count(gene_type)

pri_mouse <- as.data.frame(import("data/Gtf_Annotation/Mouse/gencode.vM19.primary_assembly.annotation.gtf"))%>% 
  dplyr::select(gene_type) %>% 
  count(gene_type)

chr_mouse <- as.data.frame(import("data/Gtf_Annotation/Mouse/gencode.vM19.annotation.gtf"))%>% 
  dplyr::select(gene_type) %>% 
  count(gene_type)


## combine dataframes to one table
table_mouse <- full_join(nlc_mouse, pri_mouse, by = 'gene_type', all=TRUE) %>%
  full_join(., chr_mouse, by='gene_type', all=TRUE)

names(table_mouse) <- c("Gene_Type", "gtf_nlc", "gtf_pri", "gtf_chr")

knitr::kable(table_mouse,
             caption = "The table shows the three different mouse annotation files with the gene types they contain")

```

## Quality Filters

```{r layout="l-page"}
fname <- "data/1820_Metrics_Summary.csv"
readMetricsSummary(fname) %>% 
  mutate(genome = "human") %>% 
  bind_rows(readMetricsSummary("data/1820_Metrics_Summary_mouse.csv") %>% mutate(genome = "mouse")) %>% 
  filter(!is.na(value)) -> metrics

metrics_reshape <- reshape(metrics,
                           idvar = "quality",
                           timevar = "genome",
                           direction = "wide")

knitr::kable(metrics_reshape,
             caption = "This metrics summary table provides statistics on sequencing, molecules, cells and targets for the 1820 sample")

```

```{r}
fname1822 <- "data/1822_Metrics_Summary.csv"
readMetricsSummary(fname1822) %>%
  mutate(genome = "human") %>%
  bind_rows(readMetricsSummary("data/1822_Metrics_Summary_mouse.csv") %>% mutate(genome = "mouse")) %>%
  filter(!is.na(value)) -> metrics1822
         

metrics1822_reshape <- reshape(metrics1822,
                            idvar = "quality",
                            timevar = "genome",
                            direction = "wide")

knitr::kable(metrics1822_reshape,
             caption = "This metrics summary table provides statistics on sequencing, molecules, cells and targets for the 1822 sample")
```


## Molecular index statistics

PCR and sequencing errors within UMI sequence generate artifact molecules and contribute to over estimation of molecules. Molecular Index (MI) adjustment identify these artifacts and discard or re-assign them to their original index (from BD slides).

```{r umiStatistics, fig.cap= "The figure shows the molecular index statistics for the 1820 sample"}
read_csv("data/1820_UMI_Adjusted_Stats.csv", comment = "##") %>% 
  mutate(genome = "human") %>% 
  bind_rows(
    read_csv("data/1820_UMI_Adjusted_Stats_mouse.csv", comment = "##") %>% 
      mutate(genome = "mouse")) -> umi.stats

umi.stats %>% 
  group_by(genome) %>% 
  summarize(meanRSEC = mean(RSEC_Adjusted_Seq_Depth_without_Singletons))

umi.stats %>% 
  gghistogram(x = "RSEC_Adjusted_Seq_Depth_without_Singletons", fill = "Status", facet.by = "genome") + 
  scale_x_log10() + 
  geom_vline(xintercept = 6, color = "darkred", lty = "dashed", size = 1)
```

```{r umiStatistics_1822, fig.cap= "The figure shows the molecular index statistics for the 1822 sample"}
read_csv("data/1822_UMI_Adjusted_Stats.csv", comment = "##") %>% 
  mutate(genome = "human") %>% 
  bind_rows(
    read_csv("data/1822_UMI_Adjusted_Stats_mouse.csv", comment = "##") %>% 
      mutate(genome = "mouse")) -> umi_1822.stats

umi_1822.stats %>% 
  group_by(genome) %>% 
  summarize(meanRSEC = mean(RSEC_Adjusted_Seq_Depth_without_Singletons))

umi_1822.stats %>% 
  gghistogram(x = "RSEC_Adjusted_Seq_Depth_without_Singletons", fill = "Status", facet.by = "genome") + 
  scale_x_log10() + 
  geom_vline(xintercept = 6, color = "darkred", lty = "dashed", size = 1)
```

In general, this is a two-step process, including RSEC-adjustment and DBEC-adjustment

1. Recursive Substitution Error Correction (RSEC)

- Corrects for MI errors that are derived from sequencing base calls and PCR substitution errors
- Based on edit distance of molecular indices
- Similar method used by most other competitors/academic pipelines

2. Distribution Based Error Correction (DBEC)

- Adjusts for errors derived from library preparation steps or sequencing base deletions.
- Based on distribution of sequencing coverage of molecular indices
- Developed at BD Genomics

Due to the low sequencing depth in our example (see threshold in Figure), DBEC-adjustment was not applied.

## Cell label filtering

The number of unique cell labels should be similar to the number of cells captured and amplified by the BD Rhapsody workflow. Causes of excess cell labels (from BD slides) :

- Spill over of mRNA to the neighboring wells
- Underloading beads in Rhapsody Cartridges: cells without beads and the RNA from the cells diffusing to adjacent wells
- Experiencing low level contamination during oligonucleotide and bead synthesis
- Generating errors during the PCR amplification steps of the workflow

In our case, the analysis of cell labels indicates two cell populations with very different expression levels.

```{r fig.cap= "Result of the second derivative analysis of the 1820 sample aligned to the human genome."}
include_graphics("plots/1820_Cell_Label_Second_Derivative_Curve_Human.png")
```

```{r fig.cap= "Result of the second derivative analysis of the 1822 sample aligned to the human genome."}
include_graphics("plots/1822_Cell_Label_Second_Derivative_Curve.png")
```

```{r fig.cap= "Result of the second derivative analysis of the 1820 sample aligned to the mouse genome."}
include_graphics("plots/1820_Cell_Label_Second_Derivative_Curve_mouse.png")
```

```{r fig.cap= "Result of the second derivative analysis of the 1822 sample aligned to the mouse genome."}
include_graphics("plots/1822_Cell_Label_Second_Derivative_Curve_mouse.png")
```

# Outline of analysis steps

We use RSEC adjusted molecules from the pipeline at 7bridges as a starting point for the analysis of human single cells and refer to [@Luecken2019] as a guideline for the analysis of the count matrices.

1. quality control

    1. number of counts per cell
    1. number of genes per cell
    1. number of mitochondrial genes per cell

1. normalization

    1. on cellular counts
    1. on gene level

1. data correction (batch / cell cycle effect ...) and integration

    1. regressing out biological effects
    1. regressing out technical effects
    1. integration of multiple samples

1. expression recovery (denoising or imputation)
1. feature selection and dimensionality reduction
1. biological downstream analysis

    1. clustering
    1. pathway analysis
    1. to be discussed ...

-> back to [top][Data overview]

# Single cell analysis
## quality control

I started with the unfiltered data sets provided by 7bridges. When creating the Seurat-Object (analysis-tool for sc-RNAsequencing), I only keep cells with at least 200 genes covered which drastically reduced the number of cells. The overall number of genes and the number of cells are shown in the table below.

```{r load-libraries}
library(Matrix)
library(Matrix.utils)
library(tidyverse)
library(Seurat)
library(janitor)
library(ggsci)
library(cowplot)
library(DT)
```

```{r load-data}
load("data/unfilteredSeurat.Rdata")
lapply(list_seurat, function(x) dim(x) %>% enframe(name = "dim")) %>% 
  bind_rows(.id = "experiment") %>% 
  mutate(dim = ifelse(dim == 1, "nrGenes", "nrCells")) %>% 
  datatable()  
mypal <- pal_jama()(2)
```

```{r set-thresholds}
nFeature.lower <- 200
nFeature.upper <- 1500
mt.upper <- 5
```

The next plots show the quality measures: number of unique genes, total molecules and the percentage of mitochondrial genes as violin plots. A high number of genes indicate possible doublets, whereas a high percentage of mitochondrial genes indicate dying cells.

```{r plot-qc, fig.keep="all"}
p1 <- VlnPlot(list_seurat[[1]], features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), 
        ncol = 3, pt.size = 0, log = TRUE, combine = FALSE, cols = mypal[1]) 
p1 <- lapply(p1, function(x){
  x + xlab(Project(list_seurat[[1]])) +
    theme(axis.text.x = element_blank(),
          legend.position = "none")
  })
p1[[1]] <- p1[[1]] + geom_hline(yintercept = c(nFeature.lower, nFeature.upper))
p1[[3]] <- p1[[3]] + geom_hline(yintercept = mt.upper)
plot_grid(plotlist = p1, nrow = 1)
p2 <- VlnPlot(list_seurat[[1]], features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), 
        ncol = 3, pt.size = 0, log = TRUE, combine = FALSE, cols = mypal[2]) 
p2 <- lapply(p2, function(x){
  x + xlab(Project(list_seurat[[2]])) +
    theme(axis.text.x = element_blank(), 
          legend.position = "none")
  })
p2[[1]] <- p2[[1]] + geom_hline(yintercept = c(nFeature.lower, nFeature.upper))
p2[[3]] <- p2[[3]] + geom_hline(yintercept = mt.upper)
plot_grid(plotlist = p2, nrow = 1)
```

```{r plot-qc-scatter}
plot1 <- FeatureScatter(list_seurat[[1]], feature1 = "nCount_RNA", feature2 = "percent.mt",
                        cols = mypal[1])
plot1 <- plot1 +
  geom_vline(xintercept = c(nFeature.lower, nFeature.upper)) +
  geom_hline(yintercept = mt.upper) + 
  scale_x_log10() + 
  theme(legend.position = "none")
plot2 <- FeatureScatter(list_seurat[[2]], feature1 = "nCount_RNA", feature2 = "percent.mt",
                        cols = mypal[2])
plot2 <- plot2 +
  geom_vline(xintercept = c(nFeature.lower, nFeature.upper)) +
  geom_hline(yintercept = mt.upper)+ 
  scale_x_log10() +
  theme(legend.position = "none")
plot1 + plot2 
```

If we use the suggested thresholds for filtering cells, we are loosing roughly 2000 to 4000 cells.
